examples/client/Locomotion/src/context/newRideContext/index.js

```diff
@@ -62,7 +64,33 @@ const RidePageContextProvider = ({ navigation, children }) => {
   const [searchResults, setSearchResults] = useState(null);
   const [isReadyForSubmit, setIsReadyForSubmit] = useState(false);
   const [historyResults, setHistoryResults] = useState([]);
-
+  const [serviceEstimations, setServiceEstimations] = useState(null);
+
+  const formatEstimations = (services, estimations) => services.map((service) => {
+    const estimationForService = estimations.find(estimation => estimation.serviceId === service.id);
+    const estimationResult = estimationForService && estimationForService.results[0];
+    return {
+      name: service.displayName,
+      eta: (estimationResult || {}).minPickupEta,
+      price: (estimationResult || {}).priceAmount,
+      availableSeats: service.maxPassengers || 4,
+      tag: null,
+      iconUrl: service.icon,
+      description: service.displayDescription,
+    };
+  });
+
+  const getServiceEstimations = async () => {
+    const formattedStopPoints = requestStopPoints.map(sp => ({
+      type: sp.type,
+      lat: sp.location.lat,
+      lng: sp.location.lng,
+    }));
+    const estimations = await createServiceEstimations(formattedStopPoints);
```

Comment:
Not sure why you need the services, maybe return them as part of the payload of estimations
But If you do need it, use promise.all

Reply:
it just seemed out of the scope of the estimations api to return all the demand sources services also, so i take the services and the estimations that were available and i create the service object needed for the ui